### ISSUE TITLE
Snowflake/Create Table: Unable to parse column with dollar quote $ in column name

### ISSUE THREAD

**Issue by myschkyna** [sqlfluff/sqlfluff#1590]
We use the dollar sign '$' to indicate that some columns are meta columns. As we have a lot of files that already contain these dollar quotes, for example but not limited to database change management files, it will become a big hassle to change all these files. We started using SQLFluff just now, and are running into linting issues for the files containing column names using dollar quotes. 

## Expected Behaviour
SQLFluff is able to parse column names containing special characters such as a dollar quote. 

## Observed Behaviour
SQLFluff can not parse the simple SQL statement for a column name containing a dollar quote. 

```bash
== [schema/test_schema/test_table.sql] FAIL
L:   2 | P:  55 |  PRS | Line 2, Position 55: Found unparsable section: '
                       | (meta$created_at TIMESTAMP_LTZ(0))'
```

## Steps to Reproduce
1. Install the SQLFluff package version 0.6.9
2. Create a file with content below
3. Run sqlfluff lint --dialect snowflake <path to file>

Content from test_table.sql:
```sql
CREATE
OR REPLACE TABLE test_schema.test_table (meta$created_at TIMESTAMP_LTZ(0));

```

## Dialect
Snowflake

## Version
sqlfluff, version 0.6.9

## Configuration
```
[sqlfluff]
dialect = snowflake
exclude_rules = L009,L011,L029,L031,L034

[sqlfluff:rules]
tab_space_size = 2

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = upper

[sqlfluff:rules:L014]  # Tables, columns, views
extended_capitalisation_policy = lower

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = upper

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = upper

[sqlfluff:rules:L016] # Max line length
max_line_length = 120
```

## Parser output
Output for SQL file that is _not_ working:

```bash
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_table_statement:
[L:  1, P:  1]      |            keyword:                                          'CREATE'
[L:  1, P:  7]      |            newline:                                          '\n'
[L:  2, P:  1]      |            binary_operator:                                  'OR'
[L:  2, P:  3]      |            whitespace:                                       ' '
[L:  2, P:  4]      |            keyword:                                          'REPLACE'
[L:  2, P: 11]      |            whitespace:                                       ' '
[L:  2, P: 12]      |            keyword:                                          'TABLE'
[L:  2, P: 17]      |            whitespace:                                       ' '
[L:  2, P: 18]      |            table_reference:
[L:  2, P: 18]      |                identifier:                                   'test_schema'
[L:  2, P: 29]      |                dot:                                          '.'
[L:  2, P: 30]      |                identifier:                                   'test_table'
[L:  2, P: 40]      |        unparsable:                                           !! Expected: 'Nothing...'
[L:  2, P: 40]      |            whitespace:                                       ' '
[L:  2, P: 41]      |            bracketed:
[L:  2, P: 41]      |                start_bracket:                                '('
[L:  2, P: 42]      |                raw:                                          'meta'
[L:  2, P: 46]      |                raw:                                          '$created_at'
[L:  2, P: 57]      |                whitespace:                                   ' '
[L:  2, P: 58]      |                raw:                                          'TIMESTAMP_LTZ'
[L:  2, P: 71]      |                bracketed:
[L:  2, P: 71]      |                    start_bracket:                            '('
[L:  2, P: 72]      |                    raw:                                      '0'
[L:  2, P: 73]      |                    end_bracket:                              ')'
[L:  2, P: 74]      |                end_bracket:                                  ')'
[L:  2, P: 75]      |    statement_terminator:                                     ';'
[L:  2, P: 76]      |    newline:                                                  '\n'

==== parsing violations ====
L:   2 | P:  40 |  PRS | Line 2, Position 40: Found unparsable section: ' (meta$created_at
                       | TIMESTAMP_LTZ(0))'
```

SQL file content that _is_ working:

```sql
CREATE
OR REPLACE TABLE test_schema.test_table (metacreated_at TIMESTAMP_LTZ(0));

```

**Comment by tunetheweb** [comment]
Hey @myschkyna you should fork the repo and open a pull request from this fork to submit this.

Always welcoming of new contributors so happy to help explain any of this if not familiar with git?

Also be aware that there's an issue with the DBT tests at the moment, so once you submit your PR don't be surprised if those tests fail - you can ignore them and we're working on fixing them.
